test/sim/team-validator/misc.js

```diff
@@ -213,4 +213,21 @@ describe('Team Validator', () => {
 		];
 		assert.legalTeam(team, 'gen8ou');
 	});
+
+	// Sometimes a Pokemon gets marked as NDZU or some such nonexistent tier on accident, resulting in it not being covered by the banlist.
+	it('should reject all Pokemon in the 35 Pokes format that are not explicitly allowed', () => {
+		const formatid = 'gen9nationaldex35pokes';
+		const rosterSize = 40;
+
+		const format = Dex.formats.get(formatid);
+		if (!format.exists) return;
+
+		const ruleTable = Dex.formats.getRuleTable(format);
+
+		const accepted = Dex.species.all().filter(species => !(
+			species.natDexTier === 'Illegal' || species.isNonstandard === 'CAP'
+		) && !ruleTable.isBannedSpecies(species)).length;
+
+		assert.equal(accepted, rosterSize);
```

Comment:
```suggestion
		const accepted = Dex.species.all().filter(species => !ruleTable.isBannedSpecies(species));
		const expectedSize = Array.from(ruleTable).filter(([index, rule]) => rule.charAt(0) === '+').length;

		assert.equal(accepted.length, expectedSize);
```
something like this?

(also you don't need to check that natDexTier === illegal or cap because isBannedSpecies should already run nonexistent checks)

Reply:
yep

> (also you don't need to check that natDexTier === illegal or cap because isBannedSpecies should already run nonexistent checks)

it doesn't, maybe it's bugged, either way not workable without the extra checks

```
> const {Dex, toID} = require('./dist/sim/index.js')
undefined
> const ruleTable = Dex.formats.getRuleTable(Dex.formats.get('gen9nationaldex35pokes'));
undefined
> Dex.species.all().filter(species => !ruleTable.isBannedSpecies(species)).length;
197
> Dex.species.all().filter(species => !(species.natDexTier === 'Illegal' || species.isNonstandard === 'CAP') && !ruleTable.isBannedSpecies(species)).length;
40
```